### DISCUSSION TITLE
Error when updating primary key

### DISCUSSION THREAD

**Discussion by femto-code** [ujjwalguptaofficial/JsStore#303] [Q&A] [answered]
Hi, I need help with an update query on one of my tables:
```
for(const el of rdata.newIds){
    var nor = await connection.update({
        in: el.type, // table
        set: {
            id: parseInt(el.newid) // old id
        },
        where: {
            id: parseInt(el.oldid) // new id to be updated
        }
    });
    console.log(`Updated ${nor} element with new ids (created)`, el);
}
```
I get this error - can someone explain what it means?

> Failed to execute 'update' on 'IDBCursor': The effective object store of this cursor uses in-line keys and evaluating the key path of the value parameter results in a different value than the cursor's effective key.

**Comment by ujjwalguptaofficial** [comment, marked as answer]
This is because of primary key update. If you need to update primary key, then keep the primary key column as unique and use another column with autoincrement for primary key.

**Reply by femto-code** [reply]
Thanks for your help! Okay, so that means there is no way to change the primary key value of an existing record. As you said, I could use a separate column. The background of my question is the following basic problem:

I want to keep a client and a server database in sync by doing regular uploads/downloads (started by user). The user can create an entry on the local systems database. When synchronized the entry gets transferred to the server and is created on the servers DB as well - with an unique ID. Up until that point the client does not know the ID that the server will assign to that record - this is needed in particular to reference it unambiguously. That is why I tried to change it afterwards (when the server tells the client the newly created records id). The reason why I cannot predict the servers ID value is that there might be another client who has created and synced one (or more) other item(s) that hold the (next) AI (autoincrement) value.

But I will think about your approach of another column holding the ID assigned by the server.
